ctms/crud.py

```diff
@@ -270,23 +270,33 @@ def get_contacts_by_any_id(
 
 
 def get_all_acoustic_records_before(
-    db: Session,
-    end_time: datetime,
-    retry_limit: int = 5,
+    db: Session, end_time: datetime, retry_limit: int = 5, batch_limit=None
 ) -> List[PendingAcousticRecord]:
     """
     Get all the pending records before a given date. Allows retry limit to be provided at query time."""
-    pending_records: List[PendingAcousticRecord] = (
+    if batch_limit is None:
```

Comment:
I think it would be clearer that the queries are similar if we move the ``batch_limit`` logic down, something like:

```python
pending_records: List[PendingAcousticRecord] = (
   db.query(PendingAcousticRecord)
   ...
)
if batch_limit:
  pending_records = pending_records.limit(batch_limit)
return pending_records.all()
```

Reply:
Did the needful, changed to reduced duplication; thanks for the suggestion